### PR TITLE
Refactor Flint Auth

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
* false is replaced with `noauth` for `glue.indexstore.opensearch.auth` property.
* Fixed few more bugs in indexstore parameters.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by YANG-DB — APPROVED** [review]

File: docs/user/ppl/admin/connectors/s3glue_connector.rst [hunk 1]
```diff
@@ -39,9 +39,8 @@ Glue Connector Properties.
     * This parameters provides the Opensearch domain host information for glue connector. This opensearch instance is used for writing index data back and also
     * ``glue.indexstore.opensearch.uri`` [Required]
     * ``glue.indexstore.opensearch.auth`` [Required]
-        * Default value for auth is ``false``.
+        * Accepted values include ["noauth", "basicauth"]
```
Comment (penghuo) [inline docs/user/ppl/admin/connectors/s3glue_connector.rst:42]: awssigv4 is also valid.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java [hunk 2]
```diff
@@ -117,19 +124,38 @@ private String constructSparkParameters(String datasourceName) {
...
+      String username =
+          dataSourceMetadata.getProperties().get(GLUE_INDEX_STORE_OPENSEARCH_AUTH_USERNAME);
+      String password =
+          dataSourceMetadata.getProperties().get(GLUE_INDEX_STORE_OPENSEARCH_AUTH_PASSWORD);
+      s3GlueSparkSubmitParameters.addParameter(FLINT_INDEX_STORE_AUTH_USERNAME, username);
+      s3GlueSparkSubmitParameters.addParameter(FLINT_INDEX_STORE_AUTH_PASSWORD, password);
+    } else if(AuthenticationType.get(authType).equals(AuthenticationType.AWSSIGV4AUTH)) {
+      s3GlueSparkSubmitParameters.addParameter(FLINT_INDEX_STORE_AUTH_KEY, "sigv4");
+    } else {
+      s3GlueSparkSubmitParameters.addParameter(FLINT_INDEX_STORE_AUTH_KEY, authType);
```
Comment (penghuo) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java:155]: what is else? "noauth"?

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java [hunk 3]
```diff
@@ -46,11 +52,34 @@ public DataSource createDataSource(DataSourceMetadata metadata) {
...
+              GLUE_INDEX_STORE_OPENSEARCH_REGION,
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH,
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH_USERNAME,
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH_PASSWORD));
+    } else if (AuthenticationType.NOAUTH.equals(authenticationType)) {
+      DatasourceValidationUtils.validateLengthAndRequiredFields(
+          dataSourceMetadataConfig,
+          Set.of(
+              GLUE_AUTH_TYPE,
+              GLUE_ROLE_ARN,
```
Comment (penghuo) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java:73]: GLUE_ROLE_ARN is not required if NOAUTH?

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java [hunk 4]
```diff
@@ -46,11 +52,34 @@ public DataSource createDataSource(DataSourceMetadata metadata) {
...
+    AuthenticationType authenticationType =
+        AuthenticationType.get(dataSourceMetadataConfig.get(GLUE_INDEX_STORE_OPENSEARCH_AUTH));
+    if (AuthenticationType.BASICAUTH.equals(authenticationType)) {
+      DatasourceValidationUtils.validateLengthAndRequiredFields(
+          dataSourceMetadataConfig,
+          Set.of(
+              GLUE_AUTH_TYPE,
+              GLUE_ROLE_ARN,
+              GLUE_INDEX_STORE_OPENSEARCH_URI,
+              GLUE_INDEX_STORE_OPENSEARCH_REGION,
```
Comment (derek-ho) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java:64]: Why do we need region for basic auth? Shouldn't region only be for sig v4?

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java [hunk 5]
```diff
@@ -46,11 +52,34 @@ public DataSource createDataSource(DataSourceMetadata metadata) {
...
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH_USERNAME,
+              GLUE_INDEX_STORE_OPENSEARCH_AUTH_PASSWORD));
+    } else if (AuthenticationType.NOAUTH.equals(authenticationType)) {
+      DatasourceValidationUtils.validateLengthAndRequiredFields(
+          dataSourceMetadataConfig,
+          Set.of(
+              GLUE_AUTH_TYPE,
+              GLUE_ROLE_ARN,
+              GLUE_INDEX_STORE_OPENSEARCH_URI,
+              GLUE_INDEX_STORE_OPENSEARCH_REGION,
```
Comment (derek-ho) [inline datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java:75]: Same question as above, why is region required for basic auth and no auth?

File: docs/user/ppl/admin/connectors/s3glue_connector.rst [hunk 6]
```diff
@@ -39,9 +39,8 @@ Glue Connector Properties.
...
     * ``glue.indexstore.opensearch.region`` [Required]
```
Comment (derek-ho) [inline docs/user/ppl/admin/connectors/s3glue_connector.rst:43]: Same question here - why is this required for basic/no auth?

File: docs/user/ppl/admin/connectors/s3glue_connector.rst (hunk 1 above)
Comment (vamsi-amazon) [reply docs/user/ppl/admin/connectors/s3glue_connector.rst:42]: You are right.
Not a patch.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java (hunk 2 above)
Comment (vamsi-amazon) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java:155]: yes

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java (hunk 3 above)
Comment (vamsi-amazon) [reply datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java:73]: There are two authentication types. one is for glue and other is for index store.
If it is NOAUTH for glue...it will throw exception only IAM_ROLE is allowed.

File: datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java (hunk 4 above)
Comment (vamsi-amazon) [reply datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java:64]: You are right. will address that.

File: docs/user/ppl/admin/connectors/s3glue_connector.rst (hunk 6 above)
Comment (vamsi-amazon) [reply docs/user/ppl/admin/connectors/s3glue_connector.rst:43]: will adress at all places.

**Review by penghuo — APPROVED** [review]

**Review by penghuo — APPROVED** [review]

**Review by ps48 — APPROVED** [review]
